signer/src/storage/mod.rs

```diff
@@ -43,6 +43,19 @@ pub trait DbRead {
         &self,
     ) -> impl Future<Output = Result<Option<model::BitcoinBlockHash>, Error>> + Send;
 
+    /// Get the latest canonical bitcoin blocks.
+    fn get_latest_canonical_bitcoin_blocks(
+        &self,
+        limit: u16,
```

Comment:
You probably want this function to take the bitcoin chain tip like most of the functions here.